Add plugin: Widget Sidebar

dmscode opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/dmscode/DMS-Widgets-Sidebar

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
Including some screenshots of the plugin UI in the README would be great.

[export class SidebarView extends ItemView {](https://github.com/dmscode/DMS-Widgets-Sidebar/blob/20e4f27807e55a752b5917b709bb9cc9552893da/src/sidebar.ts#L21)
Instead of having one view type for everything, create a view per widget
That way a user can have multiple widgets opened

["id": "dms-widget-sidebar",](https://github.com/dmscode/DMS-Widgets-Sidebar/blob/20e4f27807e55a752b5917b709bb9cc9552893da/manifest.json#L2-L3)
Don't include your name in the plugin name / id.

[https://creativecommons.org/licenses/by-nc-nd/4.0/](https://github.com/dmscode/DMS-Widgets-Sidebar/blob/20e4f27807e55a752b5917b709bb9cc9552893da/LICENSE#L1)
You need to include the license text in this file for it to be valid.

You should also add commands that activates the plugins functionality, the ribbon can be hidden and might be hidden by default in a future update.

[containerEl.createEl('h2', { text: getLang('widget_style_title') });](https://github.com/dmscode/DMS-Widgets-Sidebar/blob/20e4f27807e55a752b5917b709bb9cc9552893da/src/settings.ts#L67), [containerEl.createEl('h2', { text: getLang('widget_list_title') });](https://github.com/dmscode/DMS-Widgets-Sidebar/blob/20e4f27807e55a752b5917b709bb9cc9552893da/src/settings.ts#L84)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[ribbon_button_title: 'Widget Sidebar',](https://github.com/dmscode/DMS-Widgets-Sidebar/blob/20e4f27807e55a752b5917b709bb9cc9552893da/src/local/lang/en.ts#L3-L138)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

dmscode commented:
> Instead of having one view type for everything, create a view per widget
> That way a user can have multiple widgets opened

I don't fully understand this part. Currently, each widget is a Component, which allows control over its lifecycle and enables the creation of multiple widgets of the same type in a single sidebar. However, opening multiple sidebars with different configurations is not yet supported, as the current focus is simply to ensure the plugin runs correctly without adding complexity for now. Therefore, no additional operational commands have been included.  

I have addressed the other issues you pointed out and added screenshots of the widgets in the description.Thank you.
